### PR TITLE
fix: revamp reorg handling logic

### PR DESCRIPTION
**Pull request by PJColombo**
It requires the api changes from the following PR: [#272](https://github.com/Blobscan/blobscan/pull/272).

This PR changes the way the indexer is handling reorgs. It specifically removes the problematic reorg handling logic within the `synchronizer` and `slots-processor` components and it introduces an update that enables the indexer to now listen for `chain_reorg` SSE (Server-Sent Events) and then identifying the slots associated with the forked blocks and subsequently invoking the updated blobscan's reorg handling endpoint.

### CODE REVIEW

File: src/utils/web3.rs [hunk 1]
```diff
@@ -64,3 +66,12 @@ pub fn get_tx_versioned_hashes(tx: &Transaction) -> Result<Option<Vec<H256>>> {
         None => Ok(None),
     }
 }
+
+pub fn get_network_dencun_fork_slot(network: &Network) -> u32 {
+    match network {
+        Network::Goerli => 7413760,
```
Comment (0xGabi) [inline src/utils/web3.rs:72]:
```suggestion
    match network {
        Network::Mainnet => 8626176,
        Network::Goerli => 7413760,
```

File: src/env.rs [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
     pub execution_node_endpoint: String,
     pub secret_key: String,
-    #[serde(default = "default_dencun_fork_slot")]
-    pub dencun_fork_slot: u32,
+    pub dencun_fork_slot: Option<u32>,
     pub sentry_dsn: Option<String>,
 }
 
+fn default_network() -> Network {
+    Network::Devnet
```
Comment (0xGabi) [inline src/env.rs:22]: I wonder if we should use one of the existing network as default. And once mainnet is activated default to that.

**Review by 0xGabi — APPROVED** [review]
Well done!

I left a couple of reviews.

File: src/utils/web3.rs (hunk 1 above)
Comment (0xGabi) [reply src/utils/web3.rs:72]: We can also include mainnet because the fork slot is already decided.

File: src/env.rs (hunk 2 above)
Comment (PJColombo) [reply src/env.rs:22]: We can set mainnet as default once the upgrade happens. Right now I can't think of any reason to prefer one testnet over the other 🤔

**Pull request merged**